Upstream: unknown, waiting for void people's response
Source: https://github.com/voidlinux/void-packages/blob/1256b302a29e146814836913638aca8dc1d7b868/srcpkgs/gdb/patches/gdb-linux_nat.patch

the first chunk fixes build errors,
the non_stop part below fixes thread debugging being available
by default without setting any breakpoints or usage of libthread_db.

--- gdb-7.4.org/gdb/linux-nat.c 2013-08-10 05:24:24.651000003 +0000
+++ gdb-7.4/gdb/linux-nat.c 2013-08-10 05:25:50.966000003 +0000
@@ -71,6 +71,14 @@
 # endif
 #endif /* HAVE_PERSONALITY */
 
+#ifndef __SIGRTMIN
+#define __SIGRTMIN SIGRTMIN
+#endif
+
+#ifndef W_STOPCODE
+#define W_STOPCODE(sig) ((sig) << 8 | 0x7f)
+#endif
+
 /* This comment documents high-level logic of this file.
 
 Waiting for events in sync mode

